services/scabbard/libscabbard/src/service/shared.rs

```diff
@@ -125,12 +133,17 @@ impl ScabbardShared {
         self.accepting_batches
     }
 
-    pub fn add_batch_to_queue(&mut self, batch: BatchPair) -> Result<(), ScabbardError> {
-        self.batch_queue.push_back(batch);
+    fn update_pending_batches(&self, _batches: i64) {
```

Comment:
Please add a comment about why this requires batches to be prefixes with underscore.

Reply:
Updated with an explanation:
```rust
    /// Updates pending batches metrics gauge
    ///
    /// # Arguments
    ///
    /// * `_batches` - The number of pending batches for this service. It is prefixed with an
    /// underscore due to rust recognizing the metrics macro noop when the metrics feature is
    /// disabled
    fn update_pending_batches(&self, _batches: i64) {
        gauge!(
            "splinter.scabbard.pending_batches",
            _batches,
            "service" => format!("{}::{}", self.circuit_id, self.service_id)
        );
    } 
```